### ISSUE TITLE
Unable to get session id from payload when logging in from AIM for Macintosh

### ISSUE THREAD

**Issue by ukozi** [mk6i/retro-aim-server#10]
When logging in from AOL IM 2.01 & 4.0.9 for Macintosh (Mac OS 8.1, 68K platform) the server throws:

`time=2024-05-08T03:03:06.547Z level=DEBUG msg="accepted connection" svc=BOS ip=[my-ip]:2053
time=2024-05-08T03:03:06.548Z level=INFO msg="user session failed" svc=BOS err="unable to get session id from payload"`

I assume that where you're slicing this information out is different than what the Windows clients send because both of the aforementioned versions have also crashed every other OSCAR server I've found. Happy to provide whatever samples I can, just let me know what you need. I can also redirect the login to take place on your server if you'd like to capture the entire payload for yourself.

**Comment by mk6i** [comment]
Thanks for the report! I am keen on getting classic mac client support into RAS.

To get started, please log the login FLAP payload and share the output here (don't use any sensitive name/password credentials of course). 

You can get that payload by applying this hacky patch to the latest main branch:

```diff
diff --git a/server/oscar/connection.go b/server/oscar/connection.go
index 26418b3..c76734b 100644
--- a/server/oscar/connection.go
+++ b/server/oscar/connection.go
@@ -52,6 +52,8 @@ func (f *flapClient) SignonHandshake() (wire.FLAPSignonFrame, error) {
 	if err != nil {
 		return wire.FLAPSignonFrame{}, err
 	}
+	fmt.Printf("client login FLAP %#v\n", buf.Bytes())
+
 	flapSignonFrameIn := wire.FLAPSignonFrame{}
 	if err := wire.Unmarshal(&flapSignonFrameIn, buf); err != nil {
 		return wire.FLAPSignonFrame{}, err
```

If [building the project](https://github.com/mk6i/retro-aim-server/blob/main/docs/BUILD.md) is too onerous for you, I can commit a cleaner implementation to the next build this weekend.

**Comment by ukozi** [comment]
Here you go! 

`time=2024-05-08T15:32:42.371Z level=DEBUG msg="accepted connection" svc=BOS ip=[myip]:2054

client login FLAP []byte{0x0, 0x0, 0x0, 0x1, 0x0, 0x1, 0x0, 0x8, 0x75, 0x6b, 0x6f, 0x7a, 0x69, 0x39, 0x39, 0x39, 0x0, 0x2, 0x0, 0x6, 0xc2, 0x14, 0xb2, 0xf0, 0xc, 0xb0, 0x0, 0x9, 0x0, 0x2, 0x0, 0x15, 0x0, 0x16, 0x0, 0x2, 0x0, 0x5, 0x0, 0x1a, 0x0, 0x2, 0x2, 0x69, 0x0, 0x14, 0x0, 0x4, 0x0, 0x0, 0x0, 0xf, 0x0, 0x3, 0x0, 0x1e, 0x41, 0x49, 0x4d, 0x20, 0x32, 0x2e, 0x30, 0x31, 0x2e, 0x36, 0x31, 0x37, 0x20, 0x28, 0x4d, 0x61, 0x63, 0x20, 0x36, 0x38, 0x4b, 0x29, 0x20, 0x41, 0x49, 0x4d, 0x32, 0x4d, 0x61, 0x63, 0x0, 0x17, 0x0, 0x2, 0x0, 0x1, 0x0, 0x18, 0x0, 0x2, 0x0, 0x0, 0x0, 0x19, 0x0, 0x2, 0x0, 0x0}

time=2024-05-08T15:32:42.390Z level=INFO msg="user session failed" svc=BOS err="unable to get session id from payload"`

I have incredibly limited go experience, but can't think of a better project to expose myself to it with. Let me know if there's anything else you need!

**Comment by mk6i** [comment]
Thanks, will take a look this weekend :)

**Comment by mk6i** [comment]
Can you share the full log output with `LOG_LEVEL=debug`?

Also, the first connection should be to the auth server. What port does your client connect to? It should be configured to connect to the auth port, which is `5190` by default. If it's connecting to the `BOS` port first, that might explain this behavior.

**Comment by mk6i** [comment]
Looks like the [docs are wrong](https://github.com/mk6i/retro-aim-server/blob/main/docs/CLIENT.md). I wonder how many people got tripped up on this and abandoned the project.

```
In the Port field, enter the value of BOS_PORT found in config/settings.
```

**Comment by ukozi** [comment]
Hi there! It was cut just from this: 

```
time=2024-05-08T15:32:13.195Z level=INFO msg="starting management API server" addr=:8080
time=2024-05-08T15:32:13.196Z level=INFO msg="starting auth service" svc=AUTH addr=:5190
time=2024-05-08T15:32:13.196Z level=INFO msg="starting chat service" svc=CHAT addr=:5192
time=2024-05-08T15:32:13.196Z level=INFO msg="starting BOS service" svc=BOS addr=:5191
time=2024-05-08T15:32:13.196Z level=INFO msg="starting chat nav service" svc=CHAT_NAV addr=:5193
time=2024-05-08T15:32:13.196Z level=INFO msg="starting ALERT service" svc=ALERT addr=:5194
time=2024-05-08T15:32:42.371Z level=DEBUG msg="accepted connection" svc=BOS ip=174.161.77.210:2054
client login FLAP []byte{0x0, 0x0, 0x0, 0x1, 0x0, 0x1, 0x0, 0x8, 0x75, 0x6b, 0x6f, 0x7a, 0x69, 0x39, 0x39, 0x39, 0x0, 0x2, 0x0, 0x6, 0xc2, 0x14, 0xb2, 0xf0, 0xc, 0xb0, 0x0, 0x9, 0x0, 0x2, 0x0, 0x15, 0x0, 0x16, 0x0, 0x2, 0x0, 0x5, 0x0, 0x1a, 0x0, 0x2, 0x2, 0x69, 0x0, 0x14, 0x0, 0x4, 0x0, 0x0, 0x0, 0xf, 0x0, 0x3, 0x0, 0x1e, 0x41, 0x49, 0x4d, 0x20, 0x32, 0x2e, 0x30, 0x31, 0x2e, 0x36, 0x31, 0x37, 0x20, 0x28, 0x4d, 0x61, 0x63, 0x20, 0x36, 0x38, 0x4b, 0x29, 0x20, 0x41, 0x49, 0x4d, 0x32, 0x4d, 0x61, 0x63, 0x0, 0x17, 0x0, 0x2, 0x0, 0x1, 0x0, 0x18, 0x0, 0x2, 0x0, 0x0, 0x0, 0x19, 0x0, 0x2, 0x0, 0x0}
time=2024-05-08T15:32:42.390Z level=INFO msg="user session failed" svc=BOS err="unable to get session id from payload"

```

I installed 4 or 5 different versions of AIM on that Mac last night, rebooting, dns-sink-holing, troubleshooting every way I could. Additionally, I installed no less than 4 OSCAR server projects trying to find some combo that may work. I have **no** idea why I would have changed the port number. It must have been some input mistake on my behalf because I really can't think to why I would have done it intentionally - but sure enough, it was set to 5191. 

 
![Picture 3](https://github.com/mk6i/retro-aim-server/assets/3391773/d575d2ff-7457-4f9d-a91c-460ca67cb039)

So I fixed that back to 5190 and this gets the process further but brings back to light the behavior I saw with a different version of AIM (before I apparently went rogue with port settings) where it gets to _Finalizing_ in the connection steps, but eventually times out. I have included the entire trace here:

```
client login FLAP []byte{0x0, 0x0, 0x0, 0x1, 0x0, 0x1, 0x0, 0x5, 0x75, 0x6b, 0x6f, 0x7a, 0x69, 0x0, 0x2, 0x0, 0x6, 0xc2, 0x14, 0xb2, 0xf0, 0xc, 0xb0, 0x0, 0x9, 0x0, 0x2, 0x0, 0x15, 0x0, 0x16, 0x0, 0x2, 0x0, 0x5, 0x0, 0x1a, 0x0, 0x2, 0x2, 0x69, 0x0, 0x14, 0x0, 0x4, 0x0, 0x0, 0x0, 0xf, 0x0, 0x3, 0x0, 0x1e, 0x41, 0x49, 0x4d, 0x20, 0x32, 0x2e, 0x30, 0x31, 0x2e, 0x36, 0x31, 0x37, 0x20, 0x28, 0x4d, 0x61, 0x63, 0x20, 0x36, 0x38, 0x4b, 0x29, 0x20, 0x41, 0x49, 0x4d, 0x32, 0x4d, 0x61, 0x63, 0x0, 0x17, 0x0, 0x2, 0x0, 0x1, 0x0, 0x18, 0x0, 0x2, 0x0, 0x0, 0x0, 0x19, 0x0, 0x2, 0x0, 0x0}
time=2024-05-08T19:14:58.445Z level=DEBUG msg="accepted connection" svc=BOS ip=[my-ip]:2056
client login FLAP []byte{0x0, 0x0, 0x0, 0x1, 0x0, 0x6, 0x0, 0x24, 0x61, 0x32, 0x33, 0x32, 0x35, 0x35, 0x38, 0x36, 0x2d, 0x37, 0x62, 0x37, 0x32, 0x2d, 0x34, 0x64, 0x30, 0x32, 0x2d, 0x62, 0x37, 0x37, 0x37, 0x2d, 0x31, 0x35, 0x66, 0x34, 0x38, 0x31, 0x35, 0x66, 0x33, 0x32, 0x61, 0x37}
time=2024-05-08T19:14:58.681Z level=TRACE msg="client request -> server response" svc=BOS request.food_group=OService request.sub_group=ClientVersions request.snac_frame="{FoodGroup:1 SubGroup:23 Flags:0 RequestID:62868}" request.snac_payload="{Versions:[1 3 2 1 3 1 4 1 5 1 6 1 8 1 9 1 10 1 11 1 12 1 13 1 14 1 15 1]}" response.food_group=OService response.sub_group=OServiceHostVersions response.snac_frame="{FoodGroup:1 SubGroup:24 Flags:0 RequestID:62868}" response.snac_payload="{Versions:[1 3 2 1 3 1 4 1 5 1 6 1 8 1 9 1 10 1 11 1 12 1 13 1 14 1 15 1]}" screenName=ukozi ip=[my-ip]:2056
time=2024-05-08T19:14:58.757Z level=TRACE msg="client request -> server response" svc=BOS request.food_group=OService request.sub_group=OServiceRateParamsQuery request.snac_frame="{FoodGroup:1 SubGroup:6 Flags:0 RequestID:62869}" request.snac_payload=<nil> response.food_group=OService response.sub_group=OServiceRateParamsReply response.snac_frame="{FoodGroup:1 SubGroup:7 Flags:0 RequestID:62869}" response.snac_payload="{RateClasses:[{ID:1 WindowSize:80 ClearLevel:2500 AlertLevel:2000 LimitLevel:1500 DisconnectLevel:800 CurrentLevel:3433 MaxLevel:6000 LastTime:0 CurrentState:0}] RateGroups:[{ID:1 Pairs:[{FoodGroup:3 SubGroup:2} {FoodGroup:14 SubGroup:5} {FoodGroup:13 SubGroup:2} {FoodGroup:13 SubGroup:4} {FoodGroup:13 SubGroup:8} {FoodGroup:19 SubGroup:2} {FoodGroup:19 SubGroup:4} {FoodGroup:19 SubGroup:5} {FoodGroup:19 SubGroup:7} {FoodGroup:19 SubGroup:8} {FoodGroup:19 SubGroup:9} {FoodGroup:19 SubGroup:10} {FoodGroup:19 SubGroup:17} {FoodGroup:19 SubGroup:18} {FoodGroup:4 SubGroup:2} {FoodGroup:4 SubGroup:4} {FoodGroup:4 SubGroup:6} {FoodGroup:4 SubGroup:8} {FoodGroup:4 SubGroup:11} {FoodGroup:4 SubGroup:20} {FoodGroup:2 SubGroup:2} {FoodGroup:2 SubGroup:4} {FoodGroup:2 SubGroup:9} {FoodGroup:2 SubGroup:11} {FoodGroup:2 SubGroup:15} {FoodGroup:2 SubGroup:21} {FoodGroup:1 SubGroup:4} {FoodGroup:1 SubGroup:2} {FoodGroup:1 SubGroup:6} {FoodGroup:1 SubGroup:8} {FoodGroup:1 SubGroup:14} {FoodGroup:1 SubGroup:17} {FoodGroup:1 SubGroup:23} {FoodGroup:1 SubGroup:30} {FoodGroup:16 SubGroup:2} {FoodGroup:16 SubGroup:4}]}]}" screenName=ukozi ip=[my-ip]:2056
time=2024-05-08T19:14:58.874Z level=TRACE msg="client request" svc=BOS request.food_group=OService request.sub_group=OServiceRateParamsSubAdd request.snac_frame="{FoodGroup:1 SubGroup:8 Flags:0 RequestID:62870}" request.snac_payload={TLVRestBlock:{TLVList:[]}} screenName=ukozi ip=[my-ip]:2056
time=2024-05-08T19:14:58.973Z level=TRACE msg="client request" svc=BOS request.food_group=OService request.sub_group=OServiceIdleNotification request.snac_frame="{FoodGroup:1 SubGroup:17 Flags:0 RequestID:62871}" request.snac_payload={IdleTime:0} screenName=ukozi ip=[my-ip]:2056
time=2024-05-08T19:14:58.973Z level=TRACE msg="client request -> server response" svc=BOS request.food_group=ICBM request.sub_group=ICBMParameterQuery request.snac_frame="{FoodGroup:4 SubGroup:4 Flags:0 RequestID:62872}" request.snac_payload="{Frame:{FoodGroup:4 SubGroup:5 Flags:0 RequestID:62872} Body:{MaxSlots:100 ICBMFlags:3 MaxIncomingICBMLen:512 MaxSourceEvil:999 MaxDestinationEvil:999 MinInterICBMInterval:0}}" response.food_group=ICBM response.sub_group=ICBMParameterReply response.snac_frame="{FoodGroup:4 SubGroup:5 Flags:0 RequestID:62872}" response.snac_payload="{MaxSlots:100 ICBMFlags:3 MaxIncomingICBMLen:512 MaxSourceEvil:999 MaxDestinationEvil:999 MinInterICBMInterval:0}" screenName=ukozi ip=[my-ip]:2056
time=2024-05-08T19:14:58.974Z level=TRACE msg="client request -> server response" svc=BOS request.food_group=Locate request.sub_group=LocateRightsQuery request.snac_frame="{FoodGroup:2 SubGroup:2 Flags:0 RequestID:62873}" request.snac_payload=<nil> response.food_group=Locate response.sub_group=LocateRightsReply response.snac_frame="{FoodGroup:2 SubGroup:3 Flags:0 RequestID:62873}" response.snac_payload="{TLVRestBlock:{TLVList:[{Tag:1 Value:[3 232]} {Tag:2 Value:[3 232]} {Tag:3 Value:[3 232]} {Tag:4 Value:[3 232]} {Tag:5 Value:[3 232]}]}}" screenName=ukozi ip=[my-ip]:2056
time=2024-05-08T19:14:59.097Z level=TRACE msg="client request" svc=BOS request.food_group=ICBM request.sub_group=ICBMAddParameters request.snac_frame="{FoodGroup:4 SubGroup:2 Flags:0 RequestID:62874}" request.snac_payload="{Channel:0 ICBMFlags:0 MaxIncomingICBMLen:0 MaxSourceEvil:0 MaxDestinationEvil:0 MinInterICBMInterval:0}" screenName=ukozi ip=[my-ip]:2056
time=2024-05-08T19:14:59.193Z level=TRACE msg="client request" svc=BOS request.food_group=Locate request.sub_group=LocateSetInfo request.snac_frame="{FoodGroup:2 SubGroup:4 Flags:0 RequestID:62875}" request.snac_payload="{TLVRestBlock:{TLVList:[{Tag:1 Value:[116 101 120 116 47 97 111 108 114 116 102 59 32 99 104 97 114 115 101 116 61 34 117 115 45 97 115 99 105 105 34]} {Tag:2 Value:[0]}]}}" screenName=ukozi ip=[my-ip]:2056
time=2024-05-08T19:14:59.253Z level=TRACE msg="client request" svc=BOS request.food_group=Locate request.sub_group=LocateSetInfo request.snac_frame="{FoodGroup:2 SubGroup:4 Flags:0 RequestID:62876}" request.snac_payload="{TLVRestBlock:{TLVList:[{Tag:5 Value:[116 143 36 32 98 135 17 209 130 34 68 69 83 84 0 0 9 70 19 67 76 127 17 209 130 34 68 69 83 84 0 0]}]}}" screenName=ukozi ip=[my-ip]:2056
time=2024-05-08T19:17:57.799Z level=ERROR msg="client disconnected with error" svc=BOS err="read tcp 51.81.80.124:5191->[my-ip]:2056: read: connection reset by peer" screenName=ukozi ip=[my-ip]:2056
time=2024-05-08T19:17:57.799Z level=INFO msg="user disconnected" svc=BOS screenName=ukozi ip=[my-ip]:2056
```

**Comment by ukozi** [comment]
> Looks like the [docs are wrong](https://github.com/mk6i/retro-aim-server/blob/main/docs/CLIENT.md). I wonder how many people got tripped up on this and abandoned the project.
> 
> ```
> In the Port field, enter the value of BOS_PORT found in config/settings.
> ```

I would remember reading this and then following it and then setting it back. I really think my error in this case is my own. Maybe I accidentally overwrote the number when I leaned over the keyboard and just typed the wrong port back in due to tiredness, but - at least in my case - I don't think the docs had any say in it.

**Comment by mk6i** [comment]
This bug reminds me of the early months of the project where the client would hang because the server had not fully consumed a client request.

Let's add some logging at the beginning of the request handling flow and see if there are some request->responses that are not fully completing.

```diff
diff --git a/server/oscar/router.go b/server/oscar/router.go
index 1bfb861..cfec1be 100644
--- a/server/oscar/router.go
+++ b/server/oscar/router.go
@@ -69,5 +69,6 @@ func (rt Router) Handle(ctx context.Context, sess *state.Session, inFrame wire.S
 	if !ok {
 		return fmt.Errorf("%w. group: %d subgroup: %d", ErrRouteNotFound, inFrame.FoodGroup, inFrame.SubGroup)
 	}
+	fmt.Printf("routing request. group: %s subgroup: %s\n", wire.FoodGroupName(inFrame.FoodGroup), wire.SubGroupName(inFrame.FoodGroup, inFrame.SubGroup))
 	return h.Handle(ctx, sess, inFrame, r, rw)
 }
```

**Comment by ukozi** [comment]
```
time=2024-05-11T01:18:05.413Z level=INFO msg="starting auth service" svc=AUTH addr=:5190
time=2024-05-11T01:18:05.413Z level=INFO msg="starting management API server" addr=:8080
time=2024-05-11T01:18:05.414Z level=INFO msg="starting BOS service" svc=BOS addr=:5191
time=2024-05-11T01:18:05.414Z level=INFO msg="starting ALERT service" svc=ALERT addr=:5194
time=2024-05-11T01:18:05.414Z level=INFO msg="starting chat service" svc=CHAT addr=:5192
time=2024-05-11T01:18:05.414Z level=INFO msg="starting chat nav service" svc=CHAT_NAV addr=:5193
client login FLAP []byte{0x0, 0x0, 0x0, 0x1, 0x0, 0x1, 0x0, 0x5, 0x75, 0x6b, 0x6f, 0x7a, 0x69, 0x0, 0x2, 0x0, 0x6, 0xc2, 0x14, 0xb2, 0xf0, 0xc, 0xb0, 0x0, 0x9, 0x0, 0x2, 0x0, 0x15, 0x0, 0x16, 0x0, 0x2, 0x0, 0x5, 0x0, 0x1a, 0x0, 0x2, 0x2, 0x69, 0x0, 0x14, 0x0, 0x4, 0x0, 0x0, 0x0, 0xf, 0x0, 0x3, 0x0, 0x1e, 0x41, 0x49, 0x4d, 0x20, 0x32, 0x2e, 0x30, 0x31, 0x2e, 0x36, 0x31, 0x37, 0x20, 0x28, 0x4d, 0x61, 0x63, 0x20, 0x36, 0x38, 0x4b, 0x29, 0x20, 0x41, 0x49, 0x4d, 0x32, 0x4d, 0x61, 0x63, 0x0, 0x17, 0x0, 0x2, 0x0, 0x1, 0x0, 0x18, 0x0, 0x2, 0x0, 0x0, 0x0, 0x19, 0x0, 0x2, 0x0, 0x0}
time=2024-05-11T01:18:14.037Z level=DEBUG msg="accepted connection" svc=BOS ip=[my-ip]:2060
client login FLAP []byte{0x0, 0x0, 0x0, 0x1, 0x0, 0x6, 0x0, 0x24, 0x31, 0x31, 0x36, 0x36, 0x36, 0x34, 0x64, 0x64, 0x2d, 0x64, 0x64, 0x31, 0x62, 0x2d, 0x34, 0x63, 0x65, 0x35, 0x2d, 0x38, 0x38, 0x63, 0x36, 0x2d, 0x32, 0x31, 0x66, 0x34, 0x36, 0x63, 0x39, 0x32, 0x64, 0x34, 0x39, 0x38}
routing request. group: OService subgroup: ClientVersions
time=2024-05-11T01:18:14.228Z level=TRACE msg="client request -> server response" svc=BOS request.food_group=OService request.sub_group=ClientVersions request.snac_frame="{FoodGroup:1 SubGroup:23 Flags:0 RequestID:62886}" request.snac_payload="{Versions:[1 3 2 1 3 1 4 1 5 1 6 1 8 1 9 1 10 1 11 1 12 1 13 1 14 1 15 1]}" response.food_group=OService response.sub_group=OServiceHostVersions response.snac_frame="{FoodGroup:1 SubGroup:24 Flags:0 RequestID:62886}" response.snac_payload="{Versions:[1 3 2 1 3 1 4 1 5 1 6 1 8 1 9 1 10 1 11 1 12 1 13 1 14 1 15 1]}" screenName=ukozi ip=[my-ip]:2060
routing request. group: OService subgroup: OServiceRateParamsQuery
time=2024-05-11T01:18:14.328Z level=TRACE msg="client request -> server response" svc=BOS request.food_group=OService request.sub_group=OServiceRateParamsQuery request.snac_frame="{FoodGroup:1 SubGroup:6 Flags:0 RequestID:62887}" request.snac_payload=<nil> response.food_group=OService response.sub_group=OServiceRateParamsReply response.snac_frame="{FoodGroup:1 SubGroup:7 Flags:0 RequestID:62887}" response.snac_payload="{RateClasses:[{ID:1 WindowSize:80 ClearLevel:2500 AlertLevel:2000 LimitLevel:1500 DisconnectLevel:800 CurrentLevel:3433 MaxLevel:6000 LastTime:0 CurrentState:0}] RateGroups:[{ID:1 Pairs:[{FoodGroup:3 SubGroup:2} {FoodGroup:14 SubGroup:5} {FoodGroup:13 SubGroup:2} {FoodGroup:13 SubGroup:4} {FoodGroup:13 SubGroup:8} {FoodGroup:19 SubGroup:2} {FoodGroup:19 SubGroup:4} {FoodGroup:19 SubGroup:5} {FoodGroup:19 SubGroup:7} {FoodGroup:19 SubGroup:8} {FoodGroup:19 SubGroup:9} {FoodGroup:19 SubGroup:10} {FoodGroup:19 SubGroup:17} {FoodGroup:19 SubGroup:18} {FoodGroup:4 SubGroup:2} {FoodGroup:4 SubGroup:4} {FoodGroup:4 SubGroup:6} {FoodGroup:4 SubGroup:8} {FoodGroup:4 SubGroup:11} {FoodGroup:4 SubGroup:20} {FoodGroup:2 SubGroup:2} {FoodGroup:2 SubGroup:4} {FoodGroup:2 SubGroup:9} {FoodGroup:2 SubGroup:11} {FoodGroup:2 SubGroup:15} {FoodGroup:2 SubGroup:21} {FoodGroup:1 SubGroup:4} {FoodGroup:1 SubGroup:2} {FoodGroup:1 SubGroup:6} {FoodGroup:1 SubGroup:8} {FoodGroup:1 SubGroup:14} {FoodGroup:1 SubGroup:17} {FoodGroup:1 SubGroup:23} {FoodGroup:1 SubGroup:30} {FoodGroup:16 SubGroup:2} {FoodGroup:16 SubGroup:4}]}]}" screenName=ukozi ip=[my-ip]:2060
routing request. group: OService subgroup: OServiceRateParamsSubAdd
time=2024-05-11T01:18:14.447Z level=TRACE msg="client request" svc=BOS request.food_group=OService request.sub_group=OServiceRateParamsSubAdd request.snac_frame="{FoodGroup:1 SubGroup:8 Flags:0 RequestID:62888}" request.snac_payload={TLVRestBlock:{TLVList:[]}} screenName=ukozi ip=[my-ip]:2060
routing request. group: OService subgroup: OServiceIdleNotification
time=2024-05-11T01:18:14.550Z level=TRACE msg="client request" svc=BOS request.food_group=OService request.sub_group=OServiceIdleNotification request.snac_frame="{FoodGroup:1 SubGroup:17 Flags:0 RequestID:62889}" request.snac_payload={IdleTime:0} screenName=ukozi ip=[my-ip]:2060
routing request. group: ICBM subgroup: ICBMParameterQuery
time=2024-05-11T01:18:14.550Z level=TRACE msg="client request -> server response" svc=BOS request.food_group=ICBM request.sub_group=ICBMParameterQuery request.snac_frame="{FoodGroup:4 SubGroup:4 Flags:0 RequestID:62890}" request.snac_payload="{Frame:{FoodGroup:4 SubGroup:5 Flags:0 RequestID:62890} Body:{MaxSlots:100 ICBMFlags:3 MaxIncomingICBMLen:512 MaxSourceEvil:999 MaxDestinationEvil:999 MinInterICBMInterval:0}}" response.food_group=ICBM response.sub_group=ICBMParameterReply response.snac_frame="{FoodGroup:4 SubGroup:5 Flags:0 RequestID:62890}" response.snac_payload="{MaxSlots:100 ICBMFlags:3 MaxIncomingICBMLen:512 MaxSourceEvil:999 MaxDestinationEvil:999 MinInterICBMInterval:0}" screenName=ukozi ip=[my-ip]:2060
routing request. group: Locate subgroup: LocateRightsQuery
time=2024-05-11T01:18:14.551Z level=TRACE msg="client request -> server response" svc=BOS request.food_group=Locate request.sub_group=LocateRightsQuery request.snac_frame="{FoodGroup:2 SubGroup:2 Flags:0 RequestID:62891}" request.snac_payload=<nil> response.food_group=Locate response.sub_group=LocateRightsReply response.snac_frame="{FoodGroup:2 SubGroup:3 Flags:0 RequestID:62891}" response.snac_payload="{TLVRestBlock:{TLVList:[{Tag:1 Value:[3 232]} {Tag:2 Value:[3 232]} {Tag:3 Value:[3 232]} {Tag:4 Value:[3 232]} {Tag:5 Value:[3 232]}]}}" screenName=ukozi ip=[my-ip]:2060
routing request. group: ICBM subgroup: ICBMAddParameters
time=2024-05-11T01:18:14.678Z level=TRACE msg="client request" svc=BOS request.food_group=ICBM request.sub_group=ICBMAddParameters request.snac_frame="{FoodGroup:4 SubGroup:2 Flags:0 RequestID:62892}" request.snac_payload="{Channel:0 ICBMFlags:0 MaxIncomingICBMLen:0 MaxSourceEvil:0 MaxDestinationEvil:0 MinInterICBMInterval:0}" screenName=ukozi ip=[my-ip]:2060
routing request. group: Locate subgroup: LocateSetInfo
time=2024-05-11T01:18:14.775Z level=TRACE msg="client request" svc=BOS request.food_group=Locate request.sub_group=LocateSetInfo request.snac_frame="{FoodGroup:2 SubGroup:4 Flags:0 RequestID:62893}" request.snac_payload="{TLVRestBlock:{TLVList:[{Tag:1 Value:[116 101 120 116 47 97 111 108 114 116 102 59 32 99 104 97 114 115 101 116 61 34 117 115 45 97 115 99 105 105 34]} {Tag:2 Value:[0]}]}}" screenName=ukozi ip=[my-ip]:2060
routing request. group: Locate subgroup: LocateSetInfo
time=2024-05-11T01:18:14.776Z level=TRACE msg="client request" svc=BOS request.food_group=Locate request.sub_group=LocateSetInfo request.snac_frame="{FoodGroup:2 SubGroup:4 Flags:0 RequestID:62894}" request.snac_payload="{TLVRestBlock:{TLVList:[{Tag:5 Value:[116 143 36 32 98 135 17 209 130 34 68 69 83 84 0 0 9 70 19 67 76 127 17 209 130 34 68 69 83 84 0 0]}]}}" screenName=ukozi ip=[my-ip]:2060
time=2024-05-11T01:21:13.793Z level=ERROR msg="client disconnected with error" svc=BOS err="read tcp 51.81.80.124:5191->[my-ip]:2060: read: connection reset by peer" screenName=ukozi ip=[my-ip]:2060
time=2024-05-11T01:21:13.793Z level=INFO msg="user disconnected" svc=BOS screenName=ukozi ip=[my-ip]:2060
time=2024-05-11T01:23:27.408Z level=INFO msg="user session failed" svc=AUTH err="read tcp 51.81.80.124:5190->128.199.137.235:55342: read: connection reset by peer"
time=2024-05-11T01:23:35.409Z level=INFO msg="user session failed" svc=AUTH err="read tcp 51.81.80.124:5190->128.199.137.235:33130: read: connection reset by peer"
```

**Comment by clabland** [comment]
I'm facing a similar issue on a theoretically-supported setup.

Server: Ubuntu 20.04.4 LTS
Client: Win XP VM running AIM 4.8.2616, same issue with AIM 5.0.2829 and a version of AIM 5.1.

Running on a LAN, server at 192.168.122, client at 192.168.1.218. They can ping each other.

Server debug logs on connect:
```
time=2024-05-12T00:09:14.637Z level=INFO msg="starting chat nav service" svc=CHAT_NAV addr=:5193
time=2024-05-12T00:09:14.636Z level=INFO msg="starting management API server" addr=:8080
time=2024-05-12T00:09:14.636Z level=INFO msg="starting auth service" svc=AUTH addr=:5190
time=2024-05-12T00:09:14.636Z level=INFO msg="starting BOS service" svc=BOS addr=:5191
time=2024-05-12T00:09:14.637Z level=INFO msg="starting ALERT service" svc=ALERT addr=:5194
time=2024-05-12T00:09:14.636Z level=INFO msg="starting chat service" svc=CHAT addr=:5192
time=2024-05-12T00:09:22.190Z level=DEBUG msg="accepted connection" svc=BOS ip=192.168.1.218:2807
time=2024-05-12T00:09:22.211Z level=INFO msg="user session failed" svc=BOS err="unable to get session id from payload"
```

AIM hangs on "Verifying user name and password..."



Connection has been configured to point at server and the default 5191 BOS_PORT from the settings file:
![Screenshot 2024-05-11 181045](https://github.com/mk6i/retro-aim-server/assets/260882/01c71306-b96d-49b0-8a16-9e601d61f08e)


Relevant settings.env lines:
```
export BOS_PORT=5191
export OSCAR_HOST=192.168.1.122
```

I get the same result when running from source in https://github.com/mk6i/retro-aim-server/pull/12

Here are the logs with your patches above running from source on another Ubuntu server:
```
time=2024-05-11T18:13:20.209-06:00 level=INFO msg="starting ALERT service" svc=ALERT addr=:5194
time=2024-05-11T18:13:20.209-06:00 level=INFO msg="starting chat nav service" svc=CHAT_NAV addr=:5193
time=2024-05-11T18:13:20.209-06:00 level=INFO msg="starting BOS service" svc=BOS addr=:5191
time=2024-05-11T18:13:20.209-06:00 level=INFO msg="starting auth service" svc=AUTH addr=:5190
time=2024-05-11T18:13:20.209-06:00 level=INFO msg="starting chat service" svc=CHAT addr=:5192
time=2024-05-11T18:13:20.210-06:00 level=INFO msg="starting management API server" addr=:8080
client login FLAP []byte{0x0, 0x0, 0x0, 0x1}
time=2024-05-11T18:13:35.175-06:00 level=INFO msg="user session failed" svc=BOS err="unable to get session id from payload"
```

Hopefully I'm just missing something simple, also happy to open another issue if this is a different bug!

**Comment by mk6i** [comment]
Hello, I noticed a bug in the documentation this week. Please set the port to 5190, not 5191.

**Comment by clabland** [comment]
Ahh and of course I thought I tried that, but clearly not! It's working perfectly, on both AIM 4.8 and 5.0. Thanks for the awesome project.

**Comment by mk6i** [comment]
Excellent! Thanks for taking the time to report. 👍

**Comment by mk6i** [comment]
The originally reported issue should be fixed in #12
